Code-the-Dream-School/cc-prac-team4-front
# PP-63-Create-User-Reset-Password-page

**Pull Request Description:**

**This pull request includes the following changes:**

1. Updated the routing in the App.js file to include a dynamic route for resetting the password. The new route path is /password/reset/:token and it renders the ResetPassword component.

2. Added a new action function resetPassword in the userAction.js file to handle resetting the user's password. The function takes newPassword, confirmPassword, token, and dispatch as parameters and makes a POST request to the /api/v1/password/reset/:token endpoint with the new and confirmed passwords. It dispatches actions based on the success or failure of the request and logs the response for testing purposes.

3. Created a new ResetPassword component to handle the password reset functionality. The component uses useState and useReducer hooks for state management. It includes form submission logic, error handling, success messages, and CSS styles.

4. Added CSS styles for the ResetPassword component in the ResetPassword.module.css file. The styles define the layout and appearance of the reset password form.

**Before you start testing this pull request please ask me about the backend local environment variable in the file .env**

**Testing Steps:**

1. Run the application with these changes.
2. Verify that the routing works correctly for the /password/reset/:token path and renders the ResetPassword component.
3. Test the password reset functionality by entering valid passwords and confirming that the passwords match.
4. Verify that the success message is displayed when the password is reset successfully.
5. Verify that the error message is displayed when there is an error resetting the password.
6. Verify that the CSS styles for the ResetPassword component are applied correctly, including hover, focus, and active states for button.
![Screenshot 2023-07-07 at 2 49 37 AM](https://github.com/Code-the-Dream-School/cc-prac-team4-front/assets/2952900/8789bfdd-816e-4691-981b-ffd3e87a27a2)
